core/server/master/src/main/java/alluxio/master/metrics/MetricsStore.java

```diff
@@ -139,9 +124,33 @@ public synchronized void putClientMetrics(String hostname, String clientId,
    * @param name the metric name
    * @return the set of matched metrics
    */
-  public synchronized Set<Metric> getMetricsByInstanceTypeAndName(
-      MetricsSystem.InstanceType instanceType, String name) {
-    return getMetricsByInstanceType(instanceType).getByField(NAME_INDEX, name);
+  public Set<Metric> getMetricsByInstanceTypeAndName(MetricsSystem.InstanceType instanceType,
+      String name) {
+    if (instanceType == InstanceType.MASTER) {
+      return getMasterMetrics(name);
+    }
+
+    if (instanceType == InstanceType.WORKER) {
+      synchronized (mWorkerMetrics) {
+        return mWorkerMetrics.getByField(NAME_INDEX, name);
+      }
+    } else if (instanceType == InstanceType.CLIENT) {
+      synchronized (mWorkerMetrics) {
+        return mClientMetrics.getByField(NAME_INDEX, name);
+      }
+    } else {
+      throw new IllegalArgumentException("Unsupported instance type " + instanceType);
```

Comment:
Should `PROXY` and `CLUSTER` be supported?

Reply:
For now we do not collect and aggregate proxy metrics. And `cluster` metrics are aggregated results, so they are not handled here.